src/lib.rs

```diff
@@ -25,11 +25,6 @@ instead of one. Similarly for `memchr3`.
 #![deny(missing_docs)]
 #![doc(html_root_url = "https://docs.rs/memchr/2.0.0")]
 
-// Supporting 16-bit would be fine. If you need it, please submit a bug report
-// at https://github.com/BurntSushi/rust-memchr
-#[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
-compile_error!("memchr currently not supported on non-32 or non-64 bit");
-
 #[cfg(feature = "use_std")]
```

Comment:
It seems more apt to keep this check and instead have it include the two new widths. That way if 128 bit targets or something like that become available, we give a better error message.

```suggestion
#[cfg(not(any(target_pointer_width = "8", target_pointer_width = "16", target_pointer_width = "32", target_pointer_width = "64")))]
compile_error!("memchr currently only supports 8, 16, 32, and 64 bit targets");
#[cfg(feature = "use_std")]
```